projects/ssr-tests/src/ssr-testing.spec.ts

```diff
@@ -1,54 +1,126 @@
+        proxy = await ProxyUtils.startBackendProxyServer({
+          target: BACKEND_BASE_URL,
+        });
+        const response = await ProxyUtils.sendRequestToSsrServer(
+          REQUEST_PATH + 'not-existing-page'
+        );
+        expect(response.statusCode).toEqual(404);
+      });
+
+      it('should receive response with status 404 if HTTP error occurred when calling cms/pages API URL', async () => {
+        proxy = await ProxyUtils.startBackendProxyServer({
+          target: BACKEND_BASE_URL,
+          callback: (proxyRes, req) => {
+            if (req.url?.includes('cms/pages')) {
+              proxyRes.statusCode = 404;
+            }
+          },
+        });
+        const response = await ProxyUtils.sendRequestToSsrServer(REQUEST_PATH);
+        expect(response.statusCode).toEqual(404);
+      });
 
-  it('should receive 404 response when page is not existing', async () => {
-    proxy = await ProxyServer.startProxyServer({
-      target: BACKEND_BASE_URL,
+      it('should receive response with status 500 if HTTP error occurred when calling other than cms/pages API URL', async () => {
+        proxy = await ProxyUtils.startBackendProxyServer({
+          target: BACKEND_BASE_URL,
+          callback: (proxyRes, req) => {
+            if (req.url?.includes('cms/components')) {
+              proxyRes.statusCode = 404;
+            }
+          },
+        });
+        const response = await ProxyUtils.sendRequestToSsrServer(REQUEST_PATH);
+        expect(response.statusCode).toEqual(500);
+      });
     });
-    const response: any = await ProxyServer.sendRequest(
-      REQUEST_PATH + '/not-existing-page'
-    );
-    expect(response.statusCode).toEqual(404);
-  });
 
-  it('should receive 500 error response when a backend API returned server error', async () => {
-    proxy = await ProxyServer.startProxyServer({
-      target: BACKEND_BASE_URL,
-      throwStatus: 500,
+    describe('With caching enabled', () => {
+      beforeEach(async () => {
+        await SsrUtils.startSsrServer({ cache: true });
+      });
+
+      it('should take the response from cache for the next request if previous render succeeded', async () => {
+        proxy = await ProxyUtils.startBackendProxyServer({
+          target: BACKEND_BASE_URL,
+        });
+        let response: ProxyUtils.SsrResponse;
+        response = await ProxyUtils.sendRequestToSsrServer(REQUEST_PATH);
+        expect(response.statusCode).toEqual(200);
+
+        expectLogMessages().toContainLogs([
+          `Rendering started (${REQUEST_PATH})`,
+          `Request is waiting for the SSR rendering to complete (${REQUEST_PATH})`,
+        ]);
+
+        response = await ProxyUtils.sendRequestToSsrServer(REQUEST_PATH);
+        expect(response.statusCode).toEqual(200);
+        expectLogMessages().toContain(`Render from cache (${REQUEST_PATH})`);
+      });
+
+      it('should call rendering for the next request if previous render failed', async () => {
```

Comment:
[NITPICK]
What about:
```suggestion
      it('should render for the next request if previous render failed', async () => {
```

Reply:
```suggestion
      it('should render the next request if previous render failed', async () => {
```